src/components/SessionExpiry.jsx

```diff
@@ -0,0 +1,62 @@
+import React, {useState, useEffect} from "react";
+
+const SessionExpiry = () => {
+  const [timeLeft, setTimeLeft] = useState(600);
+  const [isReset, setIsReset] = useState(false)
+  const [isLoggedIn, setIsLoggedIn] = useState(true)
+
+  const resetTime = e => {
+    setIsReset(!isReset)
+  }
+
+  const displayTime = (timeLeft) => {
+    if(Math.floor(timeLeft % 60) < 10){
+      return Math.floor(timeLeft/60) + ":" + "0" + Math.floor(timeLeft % 60)
+    } else {
+      return Math.floor(timeLeft/60) + ":" +  Math.floor(timeLeft % 60)
+    }
+  }
+
+  useEffect(() => {
+    const countDown = () => {
+      if(isReset){
+        setTimeLeft(600)
+        setIsReset(!isReset)
+      } else {
+        setTimeLeft(timeLeft-.1)
+      }
+    }
+    if(isLoggedIn && timeLeft > 0){
+      setTimeout(() => {
```

Comment:
should this be triggering every 100 ms? in that case should it be setInterval ?

Reply:
not sure how useEffect which looks like a react lib, works.

Reply:
useEffect and useState are React Hooks.

Reply:
I think it would be better as a setTimeout because useEffect is calling the function every time the state is updated which is every time the time counts down.

Reply:
Let me try to refactor without using useEffect and using setInterval instead.